app/controllers/ruleset_controller.rb

```diff
@@ -30,38 +30,27 @@ def create
       return render @ruleset_name
     end
 
-    # Make the API calls in a begin/rescue block so that
-    # we don't crash out if the API or CLI calls fail.
-    begin
-      if @ruleset_name == 'security_ruleset' # Call 42Crunch service
-        if ENV['BYPASS_API'] == 'true'
-          crunch_results = File.read('fakecrunchresults.json')
-        else
-          crunch_results = Linters::CrunchApi.new.lint_to_json(ruleset_params[:oas_file].tempfile)
-        end
-        crunch_json = JSON.parse(crunch_results)
-        @score = crunch_json['score']
-        @openapi = crunch_json['openapiState']
-        @counter = crunch_json['issueCounter']
-        @criticality = crunch_json['criticality']
-        @issues = Array.new
-        crunch_json['data']['issues'].each do |issue|
-          lines = Array.new
-          issue[1]['issues'].each do | occurrence|
-            lines.push(occurrence['pointer'])
-          end
-          @issues.push([issue[1]['criticality'],issue[1]['description'], lines])
+    if @ruleset_name == 'security_ruleset' # Call 42Crunch service
+      if ENV['BYPASS_API'] == 'true'
+        crunch_results = File.read('fakecrunchresults.json')
+      else
+        crunch_results = Linters::CrunchApi.new(file: ruleset_params[:oas_file]).lint_to_json
+      end
+      crunch_json = JSON.parse(crunch_results)
+      @score = crunch_json['score']
```

Comment:
You can create a helper file and declare a method to handle the parsing and assigning of this json data. Or even better may be to add a new simple method that returns a hash of this data which you can then declare as an instance variable.

So rather than `lint_to_json` you could have `lint_to_hash`, and assign the data to @crunch_data (probably a bad name) and use that in the view.

Reply:
Yes, I've already started refactoring this on another branch but I didn't want to put too much in one PR